webapp/app.py

```diff
@@ -37,3 +42,101 @@ def get_tree(uri: str, branch="main"):
 @login_required
 def index(path):
     return render_template("index.html")
+
+
+@app.route('/get-users/<username>', methods=['GET'])
+@login_required
+def get_users(username: str):
+    first_name_query = """
+    query($name: String!) {
+        employees(filter: { firstName: $name }) {
+            id
+            name
+            email
+            team
+            department
+            jobTitle
+        }
+    }
+    """
+
+    last_name_query = """
+    query($name: String!) {
+        employees(filter: { surname: $name }) {
+            id
+            name
+            email
+            team
+            department
+            jobTitle
+        }
+    }
+    """
+
+    headers = {
+        "Authorization": "token " + environ.get("DIRECTORY_API_TOKEN")
+    }
+
+    # Currently directory-api only supports strict comparison of field values,
+    # so we have to send two requests instead of one for first and last names
+    first_name_response = requests.post(
+        "https://directory.wpe.internal/graphql/", json={
+            'query': first_name_query,
+            'variables': {'name': username},
+        }, headers=headers, verify=False)
+
+    last_name_response = requests.post(
+        "https://directory.wpe.internal/graphql/", json={
+            'query': last_name_query,
+            'variables': {'name': username},
+        }, headers=headers, verify=False)
+
+    if (first_name_response.status_code == 200 and
+            last_name_response.status_code == 200):
+        first_name_data = first_name_response.json().get('data', {}).get(
+            'employees', [])
+        last_name_data = last_name_response.json().get('data', {}).get(
+            'employees', [])
+        users = {emp['id']: emp for emp in first_name_data +
+                 last_name_data}.values()
+        return jsonify(list(users))
+    else:
+        return jsonify({"error": "Failed to fetch users"}), 500
+
+
+@app.route('/set-owners', methods=['POST'])
+# @login_required
```

Comment:
This is commented out just for QA purposes, it will be uncommented once ready to merge

Reply:
it doesn't have to be commented, it will be enough just to login into the app and then the endpoints will be open